new doc on HA and recommended deployments

carlaschroder opened the pull request:
Please review @karlitschek @bboule @markviens @MTRichards @cmonteroluque @msrex 

The first image is missing. I need the actual images, not hotlinks. And maybe someone could make nicer images with network diagram software?

In the "File Storage" section I don't understand "This is what separates us from WordPress or typo3." How does ownCloud compare to content management systems?

I left all comments intact in the source file; they can be removed if they're not needed anymore.

karlitschek commented:
looks really good so far :+1:

phil-davis commented on admin_manual/deployment_recommendations/deployment_recommendations.rst:
```diff
@@ -0,0 +1,513 @@
+Consider setting up a scale-out deployment, or using Federated Cloud Sharing to 
+keep individual ownCloud instances to a manageable size.
+
+.. comment: Federating instances seems the best way organically grow in an 
```
s/organically grow/to grow organically/

phil-davis commented on admin_manual/deployment_recommendations/deployment_recommendations.rst:
```diff
@@ -0,0 +1,513 @@
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+When master fails another slave can become master. Multi-master has the risk of 
+split brain and is a more complicated. Can run into deadlocks which oC tries 
```
s/is a more/is more/

phil-davis commented on admin_manual/deployment_recommendations/deployment_recommendations.rst:
```diff
@@ -0,0 +1,513 @@
+Operating System
+^^^^^^^^^^^^^^^^
+
+We are dependent on distributions that offers an easy way to install the 
```
s/offers/offer/

phil-davis commented on admin_manual/deployment_recommendations/deployment_recommendations.rst:
```diff
@@ -0,0 +1,513 @@
+.. comment: MySQL locks tables for schema updates and might even have to copy 
+   the whole table. That is pretty much a non-starter for migrations unless you 
+   are using a scale out deployment where you can apply the schema changes to 
+   each slave individually. Even the each migration might take several hours. 
```
s/Even the/Even then/

carlaschroder commented:
Thank you @phil-davis, gottem all.

carlaschroder commented:
I made this a chapter in the Admin manual, but maybe you meant for it to be a standalone whitepaper?

That's three questions now waiting for answers :)

@karlitschek @bboule @markviens @MTRichards @cmonteroluque @msrex

carlaschroder commented:
Allrightythen, one more review please @karlitschek @bboule @markviens @MTRichards @cmonteroluque @msrex
I moved it to its own directory and added a Btrfs recommendation for zero-downtime backups for small workgroups/departments.

For those who prefer PDFs, here you go:
[ownCloud_Deployment_Recommendations.pdf](https://github.com/owncloud/documentation/files/83371/ownCloud_Deployment_Recommendations.pdf)

carlaschroder commented:
In my needlessly-humble opinion this is ready to publish.

msrex commented:
The first graph is low-res (at least in the PDF), otherwise really good!

carlaschroder commented:
@msrex I copied the first graph from the original draft at https://github.com/owncloud/enterprise/wiki/OC-Deployment-Recommendations. There isn't much I can do with it, except add some spaghetti :)

carlaschroder commented:
stable8.2: 53c7aa9cd12e17123ed16c61af8046f0c19778ae
stable8.1: 5a43fe082a171182668489dffc133d37442ae745
stable8: c8cb40722475c719dd642a121017a4c1fb0e7a83

cmonteroluque commented:
@carlaschroder In the section General Recommendations two sentences are repeated. In the section Single Master DB is Single Point of Failure, Does Not Scale the last sentence is missing a subject. Otherwise, it's great.

carlaschroder commented:
Excellent, thanks @cmonteroluque

carlaschroder commented:
stable8: fcc3a00cda88c72872a21213c3c4738879be0355
stable8.1: 4ff2a18ba3b771716bc4477c2990826fc3366667
stable8.2: 02bd0994999a3cfcf010665738857da981f29741
master: 0ad0908252ed35dbc227cb63fcd01e3f9067f05d
